### PR TITLE
Small improvements to "Why the Component Model"

### PR DESCRIPTION
**Pull request by rylev**
This is just a pass over the current version to try and improve things slightly. Let me know if any particular change doesn't seem like an improvement, and I can give a justification for it. Some example changes include:
* Not conflating "compiling to WebAssembly" as "compiling for the browser"
* Being consistent with the use of "imports" and "exports" and not using alternate synonyms like "dependencies" and "outputs"

### CODE REVIEW

File: component-model/src/design/why-component-model.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
-The agreement of an interface adds a new dimension to Wasm portability. Not only are components portable across architectures and operating systems, but they are now portable across languages. A Go component can communicate directly and safely with a C or Rust component. It need not even know which language another component was written in - it needs only the component interface, expressed in WIT. Additionally, components can be linked into larger graphs, with one component satisfying another's dependencies, and deployed as units.
+The agreement of an interface adds a new dimension to Wasm portability. Not only are components portable across architectures and operating systems, but they are now portable across languages. A Go component can communicate directly and safely with a C or Rust component. It need not even know which language another component was written in - it needs only the component interface, expressed in WIT. Additionally, components can be linked into larger graphs, with one component's exports satisfying another's imports forming a new component that this the composition of the child components.
```
Comment (kate-goldenring) [inline component-model/src/design/why-component-model.md:11]: I do find the last line a little long but i am not sure how to break it out.

```suggestion
The agreement of an interface adds a new dimension to Wasm portability. Not only are components portable across architectures and operating systems, but they are now portable across languages. A Go component can communicate directly and safely with a C or Rust component. It need not even know which language another component was written in - it needs only the component interface, expressed in WIT. Additionally, components can be linked into larger graphs, with one component's exports satisfying another's imports, forming a new component that is the composition of the child components.
```

**Review by kate-goldenring — APPROVED** [review]
Thanks for putting another eye to this @rylev

File: component-model/src/design/why-component-model.md (hunk 1 above)
Comment (rylev) [reply component-model/src/design/why-component-model.md]: I just chopped off the end. Let me know what you think.

Comment (kate-goldenring) [reply component-model/src/design/why-component-model.md]: Looks good!

**Comment by rylev** [comment]
@kate-goldenring I don't have permission to merge in this repo - is there anything else I need to do here to have this merged?

**Pull request merged**